### DISCUSSION TITLE
How to use translated Spatie roles in SelectFilter

### DISCUSSION THREAD

**Discussion by rustraut** [filamentphp/filament#10930] [General]
Hello everyone,

I'm new on Filament and have trouble with SelectFilter.

I translated Spatie roles in the DB to json with Laravel-translatable, so they looks like this:

```
{"en":"User","de":"Benutzer"}
{"en":"Customer","de":"Kunde"}
```

Now in the "user" table, I want to use SelectFilter to filter users by role, to show i.e. only admin or only users or only customers.

```
SelectFilter::make('roles')
                    ->label(__('app.permission_role'))
                    ->relationship('roles', 'name')
```
                
Filter itself works fine, but the dropdown list looks as follows:

```
All
{"en":"Admin","de":"Admin"}
{"en":"User","de":"Benutzer"}
{"en":"Customer","de":"Kunde"}
```

How can I get translated strings in the dropdown?
i.e. for the German language:

```
Alle
Benutzer
Kunden
```

Thanks in advance!

**Comment by jinbatsu** [comment]
I also have this problem.
Try using this `getOptionLabelFromRecordUsing()`
https://filamentphp.com/docs/3.x/forms/fields/select#customizing-the-relationship-option-labels

```php
SelectFilter::make('roles')
   ->label(__('app.permission_role'))
   ->relationship('roles', 'name')
   ->getOptionLabelFromRecordUsing(fn($record) => $record->name)
```

But,.. the problem is, the indicator still using plain JSON string.
I'm not sure how to configure the indicator. I'm still trying to get there.
I think we could use `indicateUsing()`, and add query to convert values into label.

---Edit
It works, using this:
```php
SelectFilter::make('roles')
	->label(__('app.permission_role'))
	->relationship('roles', 'name')
	->getOptionLabelFromRecordUsing(fn($record) => $record->name)
	->indicateUsing(function (array $data): array {
		$locale = \App::getLocale();
		return Roles::select('id', 'name->'.$locale.' as name')->whereIn('id', $data['values'])->get()->pluck('name', 'id')->toArray();
	})
```

**Comment by rustraut** [comment]
Hi jinbatsu,

thank you. I have already found sloution:

https://github.com/filamentphp/filament/issues/7843

Look for the answer from @ahmedkalash at end

**Reply by jinbatsu** [reply]
Okay glad to hear, but it only fix in the Filter. Not in the indicator.
Look at this: [#9292](https://github.com/filamentphp/filament/issues/9292)
To fix that, I use `indicateUsing()`.

**Reply by rustraut** [reply]
Yes you are right.
Great thanks!
